reactioncommerce/reaction-next-starterkit
# feat: add @withCart decorator to cart and mini-cart

Resolves #189
Impact: **minor**
Type: **feature**

## Issue

Hook the cart and mini-cart components to data from GraphQL via the `@withCart` decorator.

## Solution

- Attach the decorator to the cart page component and the mini-cart component.
- Supply items to `<CartItems>` component
- Use pagination callback to enable an "infinite scroll" pagination for cart items

## Breaking changes

none

## Testing

1. With the starter-kit and reaction both running
1. Add items to the cart in the starter kit
1. Verify that Items and images show on both the cart page and in the mini-cart
1. Try to add more than 20 unique items to the cart to test that pagination works as expected

**if you're having problems getting more than 20 items in the cart, you can update the limit in `src/containers/cart/cartFragment.gql` to `items(first: 1, after: $itemsAfterCursor)` or something else.**

## Known issues

- MiniCart disappears when you click on anything in it. See ticket #202 
- Disabled the `StockWarning` because `currentQuantity` is not yet available in the GQL api. Will revisit in https://github.com/reactioncommerce/reaction-next-starterkit/issues/186 that deals with incrementing quantity in the cart.
